src/commands/link/link.ts

```diff
@@ -1,18 +1,77 @@
 import assert from 'node:assert'
 
-import { OptionValues } from 'commander'
 import inquirer from 'inquirer'
 import isEmpty from 'lodash/isEmpty.js'
+import type { NetlifyAPI } from 'netlify'
 
 import { listSites } from '../../lib/api.js'
+import { startSpinner } from '../../lib/spinner.js'
 import { chalk, logAndThrowError, exit, log, APIError } from '../../utils/command-helpers.js'
 import getRepoData from '../../utils/get-repo-data.js'
 import { ensureNetlifyIgnore } from '../../utils/gitignore.js'
 import { track } from '../../utils/telemetry/index.js'
 import type { SiteInfo } from '../../utils/types.js'
 import BaseCommand from '../base-command.js'
+import type { LinkOptionValues } from './option_values.js'
 
-const linkPrompt = async (command: BaseCommand, options: OptionValues) => {
+const findSiteByRepoUrl = async (api: NetlifyAPI, repoUrl: string): Promise<SiteInfo> => {
```

Comment:
This is extracted from below (see first commit). The behaviour is unchanged, though I did polish the output (added a spinner and changed some copy)